GTMAppAuth/Sources/AuthSession.swift

```diff
@@ -285,26 +285,32 @@ public final class AuthSession: NSObject, GTMSessionFetcherAuthorizer, NSSecureC
     }
     let callbackQueue = fetcherService?.callbackQueue ?? DispatchQueue.main
 
-    if let error = args.error, let delegate = self.delegate {
-      // If there is an updated error, use that; otherwise, use whatever is already in `args.error`
-      let newError = delegate.updatedError?(forAuthSession: self, originalError: error)
-      args.error = newError ?? error
+    let callback: () -> Void = {
+      callbackQueue.async { [weak self] in
+        guard let self = self else { return }
```

Comment:
I wonder if we should have a strong self reference in the block to retain the current AuthSession and make sure the callback is invoked. Not invoking the callback could leave the caller hanging and cause issues.

Reply:
Good point. Removed the capture list and the guard.